### PR TITLE
Tracker visible on create

### PR DESCRIPTION
**Pull request by jsandoe**
Taka,
I am so sorry. I screwed up (again) on creating the pull request. I missread the options and the effect of the last pull request was to merge the branch 'work' into the branch 'master' on my fork. I meant to merge the branch 'tracker-visible-on-create', which has the current changes. I know you are heading on vacation, but I hope I have caught you in time. 

To reiterate the description of the changes:

This PR has two main changes:

1. Restores the ability to open an existing document with assignments to pitch and time fields as they were when the document was saved,
2. Fixes bugs affecting positioning of the sonificationTracker in graphs. There were several issues. The way the graph component in CODAP decides whether an axis is a time axis varied from the plugin, so this fix aligns the plugin to the graph's rule. The initial value of the sonificationTracker global value wasn't being set in various circumstances.

I noticed that the code was no longer paying attention to user modification of the DateTime checkbox. There didn't seem to be a meaningful use case where the plugin should pay attention to this, so I made the checkbox read only.

### CODE REVIEW

File: Sonification/lib/CodapPluginHelper.js [hunk 1]
```diff
@@ -552,17 +560,22 @@ class CodapPluginHelper {
...
-            if (caseIDs.length) {
-                selectedItems = caseIDs
-                    .map(id => {
-                        // item.id is actually the case ID.
-                        return this.items[context]
-                            .find(item => item && item.id === id)
-                    });
-            } else {
-                selectedItems = this.items[context];
+            if (result.success) {
```
Comment (ttsuchiya) [inline Sonification/lib/CodapPluginHelper.js:564]: Maybe `getStrictlySelectedItems` down below could use the same treatment, where we can short circuit it with:
```js
if (!result.success) return [];
```

File: Sonification/MicroRhythm/app.js [hunk 2]
```diff
@@ -577,6 +580,20 @@ const app = new Vue({
...
+         *   * the phase and tracking global are at their minimum value
+         */
+        resetPlay() {
+            if (this.playToggle.state !== PLAY_TOGGLE_IDLE) this.playToggle.state = PLAY_TOGGLE_IDLE;
+            this.stop();
+            this.phase = 0;
+            let timeAdj = this.state.timeAttrIsDate? 1000: 1;
+            let trackerMin = this.timeAttrRange? this.timeAttrRange.min/timeAdj: 0;
+            helper.setGlobal(trackingGlobalName, trackerMin);
+        },
```
Comment (ttsuchiya) [inline Sonification/MicroRhythm/app.js:596]: Thanks for cleaning up the repeated code!

File: Sonification/MicroRhythm/app.js [hunk 3]
```diff
@@ -315,16 +310,23 @@ const app = new Vue({
...
-            // an attribute is a Date attribute if none of its values are non-dates
-            let isDateAttribute = !values.some((x) => {
-                let isDate = (x instanceof Date) ||
-                    ((typeof x === 'string') && !isNaN(new Date(x)));
-                return !isDate;
-            });
+            let attrType = this.getAttributeType(attrName);
+            let values = helper.getAttrValuesForContext(contextName, attrName) || [];
+            // an attribute is a Date attribute if attribute type is 'date' or
+            // all of its values are Date objects or date strings
```
Comment (ttsuchiya) [inline Sonification/MicroRhythm/app.js:323]: 👍

**Review by ttsuchiya — APPROVED** [review]
Overall looks good to me! Thanks

File: Sonification/MicroRhythm/index.html [hunk 4]
```diff
@@ -74,7 +74,7 @@
 
             <div>
                 <label title="Whether the attribute controlling pitch has date/time values or is numeric">
-                    <input type="checkbox" v-model="state.pitchAttrIsDate" v-on:change="onPitchAttributeSelectedByUI"> DateTime
+                    <input type="checkbox" v-model="state.pitchAttrIsDate" disabled="true"  v-on:change="onPitchAttributeSelectedByUI"> DateTime
```
Comment (ttsuchiya) [inline Sonification/MicroRhythm/index.html:77]: 👍

File: Sonification/lib/CodapPluginHelper.js (hunk 1 above)
Comment (jsandoe) [reply Sonification/lib/CodapPluginHelper.js:564]: Seems like a good idea! I added it to `getStrictlySelectedItems`

**Pull request merged**